ecc_enclave/enclave/shim.cpp

```diff
@@ -63,7 +63,7 @@ void get_state(
     if (check_cmac(key, NULL, &state_hash, &session_key, &cmac) != 0)
     {
         LOG_ERROR("Enclave: VIOLATION!!! Oh oh! cmac does not match!");
-        // TODO: proper error handling/abort rather than just continue
+        throw std::runtime_error("Enclave: VIOLATION!!! Oh oh! cmac does not match!");
```

Comment:
aha! I guess this will break the mock server as this does not have a tlcc instance.

Reply:
yep, bruno and me also talked about that.  However, instead of just ignoring the errors (unconditionally!) as before, i think the answer is here for the mock-server to provide a mock-tlc and just just fake the cmacs?

Reply:
I can look into that mock-tlcc tmrw.

Reply:
actually, after further reflection. I think i will undo (comment out + TODO) the exceptions for now.   Spending time on a mock-tlcc doesn't seem time well spent.  I think we should at some point add some overall/project-wide build variable MOCK/INSECURE to bracket such absence of checks as well as bracket hard-coded keys and alike.  But that we maybe can do once we start upgrading to fabric 2.0 and start the refactoring to the new design for earnest ...

Reply:
👍